### DISCUSSION TITLE
Go mod download private git registry.

### DISCUSSION THREAD

**Discussion by lichunqiang** [crazy-max/goxx#72] [Q&A] [answered]
My project depends on a private Git registry. When using Goxx to cross-build, `go mod download` fails.

What's the best practice for this scenario?

**Comment by crazy-max** [comment, marked as answer]
I don't think this is relevant to goxx but your workflow as you would have the same issue when building with standard golang image but you can do smth similar to https://github.com/docker/build-push-action/issues/1112#issuecomment-2098204816 using a [secret mount](https://docs.docker.com/build/building/secrets/#secret-mounts).

In your case:

```dockerfile
RUN --mount=type=secret,id=GIT_AUTH_TOKEN \
  set -e
  GIT_AUTH_TOKEN=$(cat /run/secrets/GIT_AUTH_TOKEN)
  if [ -n "$GIT_AUTH_TOKEN" ]; then
    echo "Setting GitHub access token"
    git config --global "url.https://x-access-token:${GIT_AUTH_TOKEN}@github.com.insteadof" "https://github.com"
  fi
  go mod download -x
EOT
```
